### ISSUE TITLE
UIPageViewController

### ISSUE THREAD

**Issue by tbilopavlovic** [RxSwiftCommunity/RxFlow#100]
Hi, are there any suggestions how to use flows with UIPageViewController? 
Currently (develop branch), `rxDismissed ` is called and flow does not listen for steps any more (just scroll to the end to get `rxDismissed` called and then scroll back to first view).
For example DashboardFlow from Demo app:

```
let rootViewController = UIPageViewController(transitionStyle: .scroll, navigationOrientation: .horizontal, options: nil)
var vcs: [UIViewController] = []
private func navigateToDashboard() -> FlowContributors {
        let wishlistStepper = WishlistStepper(preferencesServices: self.services.preferencesService)
        let wishlistStepper2 = WishlistStepper(preferencesServices: self.services.preferencesService)
        let wishlistStepper3 = WishlistStepper(preferencesServices: self.services.preferencesService)
        let wishlistStepper4 = WishlistStepper(preferencesServices: self.services.preferencesService)
        let wishlistStepper5 = WishlistStepper(preferencesServices: self.services.preferencesService)

        let wishListFlow = WishlistFlow(withServices: self.services, andStepper: wishlistStepper)
        let wishListFlow2 = WishlistFlow(withServices: self.services, andStepper: wishlistStepper2)
        let wishListFlow3 = WishlistFlow(withServices: self.services, andStepper: wishlistStepper3)
        let wishListFlow4 = WishlistFlow(withServices: self.services, andStepper: wishlistStepper4)
        let wishListFlow5 = WishlistFlow(withServices: self.services, andStepper: wishlistStepper5)
        let watchedFlow = WatchedFlow(withServices: self.services)

        Flows.whenReady(flows: [wishListFlow, wishListFlow2, wishListFlow3, wishListFlow4, wishListFlow5, watchedFlow]) {[unowned self](roots) in
            roots[0].title = "Wishlist"
            roots[1].title = "Watched"
            roots[2].title = "Wishlist"
            roots[3].title = "Wishlist"
            roots[4].title = "Wishlist"
            roots[5].title = "Wishlist"
            self.vcs = [roots[0], roots[1], roots[2], roots[3], roots[4], roots[5]]
            self.rootViewController.setViewControllers([roots[0]], direction: .forward, animated: false, completion: nil)
        }

        return .multiple(flowContributors: [.contribute(withNextPresentable: wishListFlow,
                                                        withNextStepper: wishlistStepper),
                                            .contribute(withNextPresentable: wishListFlow2,
                                                        withNextStepper: wishlistStepper2),
                                            .contribute(withNextPresentable: wishListFlow3,
                                                        withNextStepper: wishlistStepper3),
                                            .contribute(withNextPresentable: wishListFlow4,
                                                        withNextStepper: wishlistStepper4),
                                            .contribute(withNextPresentable: wishListFlow5,
                                                        withNextStepper: wishlistStepper5),
                                            .contribute(withNextPresentable: watchedFlow,
                                                        withNextStepper: OneStepper(withSingleStep: DemoStep.movieList))])
    }
}

extension DashboardFlow: UIPageViewControllerDataSource {
    func pageViewController(_ pageViewController: UIPageViewController, viewControllerAfter viewController: UIViewController) -> UIViewController? {
        if let current = vcs.enumerated().filter({ $0.element === viewController }).first?.offset, vcs.indices.contains(current + 1) {
            return vcs[current + 1]
        }
        return nil
    }
    
    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
        if let current = vcs.enumerated().filter({ $0.element === viewController }).first?.offset, vcs.indices.contains(current - 1) {
            return vcs[current - 1]
        }
        return nil
    }
}```

**Comment by twittemb** [comment]
Hi @tbilopavlovic 

Do you really want to have a navigation stack per page ?

**Comment by tbilopavlovic** [comment]
Hi,

Well each page have multiple navigation options, so i am using it as root of each new flow... but it seems i'll have to look for other solutions?

Thanks

**Comment by twittemb** [comment]
I have to try by myself to give you a sharper answer. 

Are you sure that the VCs are dismissed each time you swipe to another page ?

**Comment by tbilopavlovic** [comment]
Yeah, `rxDismissed` is triggered and flow deinit is called.

**Comment by MortyMerr** [comment]
Hi, i have the same problem. Attached sample project here - https://github.com/RxSwiftCommunity/RxFlow/issues/117

**Comment by twittemb** [comment]
This should have been addressed in the release 2.11.0 https://github.com/RxSwiftCommunity/RxFlow/releases/tag/2.11.0.
The demo app has been upgraded with a UIPageController example.
I close the issue. Feel free to reopen if you don't think it has been addressed.
